newswires/client/src/context/SearchReducer.ts

```diff
@@ -1,27 +1,49 @@
 import dateMath from '@elastic/datemath';
 import { isEqual as deepIsEqual } from 'lodash';
-import moment from 'moment';
-import type { Query, WiresQueryResponse } from '../sharedTypes.ts';
+import moment from 'moment-timezone';
+import { convertToLocalDate } from '../dateHelpers.ts';
+import type { Query, WireData, WiresQueryResponse } from '../sharedTypes.ts';
 import { defaultQuery } from '../urlState.ts';
 import type { Action, SearchHistory, State } from './SearchContext.tsx';
 
+export const safeReducer = (
+	reducer: (state: State, action: Action) => State,
+) => {
+	return (state: State, action: Action): State => {
+		try {
+			return reducer(state, action);
+		} catch (error: unknown) {
+			return {
+				...state,
+				status: 'error',
+				error: error instanceof Error ? error.message : 'Unknown error',
```

Comment:
```suggestion
				error: `Error processing action (${action.type}) ${error instanceof Error ? error.message : 'Unknown error'}`,
```

good to add a bit more context?